### PR TITLE
alot: update resources, fix build on ARM64/Apple Silicon/non-default prefixes

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This became a lot more involved than planned 🙃. I need to report the `notmuch` issue to `notmuch` upstream but I'd like some CI validation first this works on non-ARM machines just fine with the change too.

### CODE REVIEW

File: Formula/alot.rb [hunk 1]
```diff
@@ -43,54 +53,59 @@ class Alot < Formula
     sha256 "586372eb92059873e29eba4f9dec8381541b4d3834660707faf8ba59146dfc35"
   end
 
+  resource "gpg" do
+    url "https://files.pythonhosted.org/packages/ef/86/c5a34243a932346c59cb25eb49a4d1dec227974209eb9b618d0ed57ea5be/gpg-1.10.0.tar.gz"
+    sha256 "349214a866c84aa548bc35ed14eccd2ec9085b3958d5753a63a19a71a1f523ca"
+  end
```
Comment (Bo98) [inline Formula/alot.rb:59]: This is provided by `depends_on "gpgme"` I think, and the formula has a newer version.

File: Formula/alot.rb [hunk 2]
```diff
@@ -43,54 +53,59 @@ class Alot < Formula
...
+    sha256 "7d3fbbde18228f4ff2f1f119a45cdffa458b4c0dee32eb4d2bb2f82554bac7bc"
   end
 
-  resource "python-magic" do
-    url "https://files.pythonhosted.org/packages/84/30/80932401906eaf787f2e9bd86dc458f1d2e75b064b4c187341f29516945c/python-magic-0.4.15.tar.gz"
-    sha256 "f3765c0f582d2dfc72c15f3b5a82aecfae9498bd29ca840d72f37d7bd38bfcd5"
+  resource "notmuch" do
+    url "https://files.pythonhosted.org/packages/89/df/4d3618124944b2939c5cc4a35b37752cb0b76f5e198a84026108a3b4f7eb/notmuch-0.29.2.tar.gz"
+    sha256 "e32d04340780110a9c50f5cf7e0eed81b47a3785971d02dd6d99acb3f6286ad5"
   end
```
Comment (Bo98) [inline Formula/alot.rb:84]: Similarly, we have `depends_on "notmuch"` already which should have the bindings I think.

File: Formula/alot.rb (hunk 1 above)
Comment (DomT4) [reply Formula/alot.rb:59]: 👍🏻 I used the same tool we used to use last time I was involved in Homebrew so I suspect my dependency update methodology here was a little outdated. I'll change this.

File: Formula/alot.rb (hunk 2 above)
Comment (DomT4) [reply Formula/alot.rb:84]: It does, yeah. I wasn't sure if it was preferably to vendor `notmuch`'s bindings so we could keep the `inreplace` local or whether it was better to simply swallow the extra revision bump and fix `notmuch` itself.

File: Formula/alot.rb (hunk 1 above)
Comment (Bo98) [reply Formula/alot.rb:59]: We have a `brew update-python-resources` now! Might need to adjust `pypi_formula_mappings.json` though to give `brew update-python-resources` a hint that it should skip it.

File: Formula/alot.rb (hunk 2 above)
Comment (Bo98) [reply Formula/alot.rb:84]: Sounds like a bug that would affect anyone using the bindings in the `notmuch` formula so it's probably worth fixing there.

File: Formula/alot.rb (hunk 1 above)
Comment (DomT4) [reply Formula/alot.rb:59]: I tweaked `pypi_formula_mappings.json` in a way that sort of made sense to me after I spotted it but you may need to nudge me on what needs changing further if that wasn't sufficient.

File: Formula/alot.rb (hunk 2 above)
Comment (DomT4) [reply Formula/alot.rb:84]: No worries, I'll tweak the PR to do that shortly. I need to report it upstream anyway so good incentive I guess heh.

File: Formula/alot.rb (hunk 1 above)
Comment (Bo98) [reply Formula/alot.rb:59]: Ths JSON file is solely for `brew update-python-resources` so it's just a case of tweaking it if the result of that command is not what you expect.

I just realised however that `alot` isn't on PyPI so my suggestion perhaps won't work, in which case you don't need to touch the JSON at all. I'm surprised `homebrew-pypi-poet` works (assuming that's what you were using).

Comment (DomT4) [reply Formula/alot.rb:59]: >  it's just a case of tweaking it if the list of resources from that command is not what you expect.

I suspect it'd leave `wheel` out as non-essential. I noticed from the `json` file that's a bit of a theme throughout.

> I'm surprised homebrew-pypi-poet works in that case (assuming that's what you were using).

It was indeed!

```
cd /tmp
mkdir venv && cd venv
bin/pip install https://github.com/pazz/alot/archive/refs/tags/0.9.1.tar.gz
bin/pip install homebrew-pypi-poet
bin/poet alot
```

Comment (Bo98) [reply Formula/alot.rb:59]: I suppose it's a slight shortcoming of `brew update-python-resources` then. It doesn't download/install anything and just relies on PyPI searching - which won't work if it's not on PyPI!

> I suspect it'd leave `wheel` out as non-essential. I noticed from the `json` file that's a bit of a theme throughout.

Yes, there was an issue where this was the case. Think it was fixed recently in https://github.com/Homebrew/brew/pull/10849.

**Comment by DomT4** [comment]
Emailed the bug report and a suggested patch to upstream; waiting for it to be approved on the mailing list so I can link it in the formula, then hopefully this should be ready to get ARM-tested by BTB and merged in if everyone human & robot is happy 🤞🏻.

**Comment by DomT4** [comment]
Still waiting on upstream. Getting the feeling despite saying I didn't have to on the website I'm going to end up subscribed to this mailing list to get them to notice the patch/bug report 🙃.

**Comment by DomT4** [comment]
Assuming it passes CI and you're happy this should be good to go @Bo98. Ended up subscribing to the upstream mailing list, what a surprise 🙃.

**Review by Bo98 — APPROVED** [review]
Thanks! ❤️

**Comment by DomT4** [comment]
Thanks @Bo98 ❤️. I think you should be able to get the test bot to generate an ARM64 bottle for `alot` now if you wanted to

**Comment by Bo98** [comment]
Ah, yes. I perhaps should have labelled `CI-force-arm` here.

**Comment by DomT4** [comment]
I didn't realise until it'd been merged either to be fair 😬. I assume you can get the CI to do it behind the scenes still? We used to be able to do manual bottle jobs.

**Comment by Bo98** [comment]
Yep. `brew dispatch-build-bottle --macos=11-arm64 --upload alot`

https://github.com/Homebrew/homebrew-core/runs/2458404846?check_suite_focus=true

**Comment by DomT4** [comment]
The syntax for everything has changed, I am a dinosaur 😸. Thanks for sorting that so quickly!

**Pull request closed without merging**